src/main.ts

```diff
@@ -28,15 +28,31 @@ export async function readPackageFile(packagePath: string): Promise<PackageFile>
 async function run(): Promise<void> {
     try {
         const packagePath = core.getInput('path') || '.'
+        const allowFirstVersion = core.getInput('allow-first-version') === 'true'
         const packageFile = await readPackageFile(packagePath)
         core.debug(`Fetching package ${packageFile.name} information from npm…`)
-        const packageNpm = await packageJson(packageFile.name, { allVersions: true })
-        const isNewVersion = !Object.keys(packageNpm.versions).includes(packageFile.version)
-        core.setOutput('is-new-version', isNewVersion.toString())
-        core.setOutput('published-version', packageNpm['dist-tags'].latest)
-        core.setOutput('committed-version', packageFile.version)
-    } catch (error) {
-        core.setFailed(error.message)
+        try {
+            const packageNpm = await packageJson(packageFile.name, { allVersions: true })
+            const isNewVersion = !Object.keys(packageNpm.versions).includes(packageFile.version)
```

Comment:
```suggestion
            const isNewVersion = !Object.hasOwn(packageNpm.versions, packageFile.version)
```
this is O(1), instead of O(n)